### PR TITLE
Add a PageConfigWriter that writes the metadata for users of BasicSubscription.

### PR DESCRIPTION
**Pull request by stellachui**
Add a PageConfigWriter that writes the metadata specified in BasicSubscriptions' init into the HTML of the page. Implements the configuration within basic-runtime.com.

### CODE REVIEW

File: src/model/page-config-writer-test.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            '@type': ['CreativeWork', 'Product'],
+          },
+        })
+      );
+      await expect(configWriter.writeConfigWhenReady(markupConfig)).to.be
+        .eventually.fulfilled;
+      expect(gd.getHead().childNodes).to.have.length(1);
+    });
+
+    it('should write a new node if there is an existing but invalid schema', async () => {
```
Comment (jpettitt) [inline src/model/page-config-writer-test.js:110]: Type should be additive so `@type: "foobar"` would become `@type: [ "foobar", "CreativeWork", "Product" ]` in the json.   Needs to handle both string and array `@type: "foobar"` and `@type: ["foobar"]`

**Review by jpettitt — COMMENTED** [review]
Added a comment in the test because it seemed to be the best place to capture the change needed to the writer.

File: src/model/page-config-writer-test.js (hunk 1 above)
Comment (stellachui) [reply src/model/page-config-writer-test.js]: Done, I've made both the @type and the isPartOf[@type] fields take either a string or an array of strings, and merged those fields in the PageConfigWriter.

**Comment by stellachui** [comment]
Updated behavior based on feedback, and added more tests to cover the new behavior. PTAL, thanks!

**Review by jpettitt — APPROVED** [review]

**Pull request merged**